core/model/modx/sources/modfilemediasource.class.php

```diff
@@ -494,6 +494,21 @@ public function renameObject($oldPath,$newName) {
             return false;
         }
 
+        // Check that the new file has a file type that is allowed
+        $allowedFileTypes = explode(',',$this->xpdo->getOption('upload_files',null,''));
+        $allowedFileTypes = array_merge(explode(',',$this->xpdo->getOption('upload_images')),explode(',',$this->xpdo->getOption('upload_media')),explode(',',$this->xpdo->getOption('upload_flash')),$allowedFileTypes);
+        $allowedFileTypes = array_unique($allowedFileTypes);
+
+        $file_split = explode('.', $newName);
+        $ext = $file_split[count($file_split) - 1];
```

Comment:
For about 15 years, this has been my goto way of grabbing the file extension. I was not aware of the `pathinfo()` approach, but I found it used in the same class, here: https://github.com/OptimusCrime/revolution/blob/26f59a39e238333d5185161c4c1f7d8b74610c70/core/model/modx/sources/modfilemediasource.class.php#L726

Do you want me to rewrite it? Pretty sure the implementation is more or less identical under the hood, but doing things one way may be better?